File: Pong/src/model/Enemy.java
package model;

import util.AABB;
import util.Vector2;
import java.util.*;

public class Enemy extends Entity
{
    Vector2 direction = new Vector2(0f, 0f);

    private Entity ball;
    private Entity window;

    public Enemy()
    {
        super(new AABB(800-64f, 600f/2 - 64, 16f, 128f), "Enemy");
    }

    public void ready()
    {
        ball = getEntity("Ball");
        window = getEntity("Window");

        rect.x = window.rect.w - 64;
    }

    public void update()
    {
        direction.y = -1f;

        if (rect.y <= 0)
        {
            rect.y = 0f;
        }
        if (rect.y + rect.h >= window.rect.h)
        {
            rect.y = window.rect.h - rect.h;
        }

        float speed = 7f;
        velocity = direction.scale(speed);

        super.update();
    }
}

File: Pong/src/model/Enemy.kt
package model

import Window
import util.AABB
import util.Vector2

class Enemy : Entity(AABB(800-64f, 600f/2 - 64, 16f, 128f), "Enemy")
{
    var direction: Vector2 = Vector2(0f, 0f)
    private val speed = 12f
    private lateinit var ball: Ball
    private lateinit var window: Window


    override fun ready()
    {
        ball = getEntity("Ball") as Ball
        window = getEntity("Window") as Window

        rect.x = window.rect.w - 64

    }

    override fun update()
    {
        direction.y = ball.velocity.y
        direction = direction.normalize()
        //velocity = direction * speed
        super.update()
    }
}
